KPP/Hg/gckpp_Global.F90

```diff
@@ -29,54 +32,42 @@ MODULE gckpp_Global
 
 ! Declaration of global variables
 
-! These quantities vary with lon/lat/lev location.
-! They must be declared THREADPRIVATE for the parallel
-! loop over all cells in the chemistry grid.
+! ~~~ If you are using KPP within an OpenMP parallel environment,
+! ~~~ then these variables must be declared THREADPRIVATE.  This means
+! ~~~ that the compiler will make a private copy of these variables
+! ~~~ (in stack memory) for each execution thread.  At the end of
+! ~~~ the OpenMP parallel loop, these variables will be finalized,
+! ~~~ and their memory deallocated.
+! ~~~
+! ~~~ NOTE: Because the OpenMP commands all begin with a comment
+! ~~~ character, they will be ignored unless the code is compiled
+! ~~~ with OpenMP parallelization turned on.
 
 ! C - Concentration of all species
-  REAL(kind=dp) :: C(NSPEC)
+  REAL(kind=dp), TARGET :: C(NSPEC)
   !$OMP THREADPRIVATE( C )
 ! VAR - Concentrations of variable species (global)
-  REAL(kind=dp) :: VAR(NVAR)
+  REAL(kind=dp), POINTER :: VAR(:)
   !$OMP THREADPRIVATE( VAR )
 ! FIX - Concentrations of fixed species (global)
-  REAL(kind=dp) :: FIX(NFIX)
+  REAL(kind=dp), POINTER :: FIX(:)
```

Comment:
KPP 2.5.0 now automatically generates the gckpp_Global.F90 file with C as a target and VAR and FIX as pointers.  This gets rid of the non-thread-safe EQUIVALENCE statement, and makes the code thread-safe for OpenMP.